### PR TITLE
Adding support to configure the VM against environment

### PR DESCRIPTION
**Pull request by chshrikh**
this PR will enable the support to add the vm resource to environment

### CODE REVIEW

**Comment by zachariahcox** [comment]
Hi guys! Please provide some links to the feature spec. We may need a meeting with @jtpetty to walk us through what we're doing here.

**Comment by chshrikh** [comment]
@zachariahcox I have shared the required spec details in mail, I will set-up some time to discuss this with you guys

File: src/Agent.Listener/Configuration/ConfigurationProvider.cs [hunk 1]
```diff
@@ -367,4 +366,176 @@ public override async Task GetPoolIdAndName(AgentSettings agentSettings, Command
...
+
+            var environmentInstance = await GetEnvironmentAsync(_projectName, environmentName);
+           
+            agentSettings.EnvironmentId = environmentInstance.Id;
+            agentSettings.ProjectName = environmentInstance.Project.Name;
+            agentSettings.ProjectId = environmentInstance.Project.Id.ToString();            
+            Trace.Info("vm resource configuration: environment id: '{0}', project name: '{1}', project id: '{2}'", agentSettings.EnvironmentId, agentSettings.ProjectName, agentSettings.ProjectId);
+        }
+
+        public override string GetFailedToFindPoolErrorString() => StringUtil.Loc("FailedToFindEnvironment");
```
Comment (jtpetty) [inline src/Agent.Listener/Configuration/ConfigurationProvider.cs]: is it worth including the environment name in this error string?

Comment (chshrikh) [reply src/Agent.Listener/Configuration/ConfigurationProvider.cs]: yes, this will hep us in future to differentiate the error message.

**Pull request merged**